File: public/extract_pdf.py
import sys
import subprocess
try:
    import pypdf
except ImportError:
    subprocess.check_call([sys.executable, "-m", "pip", "install", "pypdf"])
    import pypdf

reader = pypdf.PdfReader("/Users/weihong/Documents/profolio/public/resume.pdf")
text = ""
for page in reader.pages:
    text += page.extract_text() + "\n"

with open("/Users/weihong/Documents/profolio/public/resume-text.txt", "w") as f:
    f.write(text)
print("Extracted to resume-text.txt")

File: public/extract_pdf.js
const fs = require('fs');
const pdf = require('pdf-parse');
let dataBuffer = fs.readFileSync('/Users/weihong/Documents/profolio/public/resume.pdf');
pdf(dataBuffer).then(function(data) {
    fs.writeFileSync('/Users/weihong/Documents/profolio/public/resume-text.txt', data.text);
    console.log("Extracted to resume-text.txt");
}).catch(console.error);
